packages/yoastseo/src/parse/structure/SourceCodeLocation.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * Represents a location of a node in the HTML tree.
+ *
+ * @param {Object?} startTag The location of the start tag of the element, if it has one.
+ * @param {number} startTag.startOffset The start position of the start tag.
+ * @param {number} startTag.endOffset The end position of the start tag.
+ *
+ * @param {Object?} endTag The location of the end tag of the element, if it has one.
+ * @param {number} endTag.startOffset The start position of the end tag.
+ * @param {number} endTag.endOffset The end position of the end tag.
+ *
+ * @param {Object} startOffset The start position of the element.
+ * @param {Object} endOffset The end position of the element.
+ */
+class SourceCodeLocation {
+	/**
+	 * Creates a new SourceCodeLocation.
+	 *
+	 * @param {Object} sourceCodeLocationInfo This node's location in the source code, from parse5.
+	 */
+	constructor( sourceCodeLocationInfo ) {
+		if ( sourceCodeLocationInfo.startTag ) {
+			this.startTag = {
+				startOffset: sourceCodeLocationInfo.startTag.startOffset,
+				endOffset: sourceCodeLocationInfo.startTag.endOffset,
+			};
```

Comment:
I'm wondering if we should introduce a `Range` class? To represent a range between two numbers.

We can use it for the start- and end tag locations and possibly the start- and end position of the entire element. And also for in the `Mark` to represent the range of text that should be marked.

Reply:
Good suggestion! I'm a bit on the fence though, I do like the idea but I am wondering if we are not overengineering -- do you have a use case in mind? 😄 If so, I can already create functions for that model 👍

Reply:
Mostly to add a name or type to an object, instead of relying on anonymous objects. It can also provide a surface for methods like computing the length of the range. 

But I agree that it may be overengineering.

For this Typescript's `type` would have been ideal, since it is more lightweight than a full blown class 😄. Maybe we can introduce a JSDoc `@typedef` instead, and use that?

Reply:
OK, let's go the `@typedef` route for now! 👍 We can always expand later. Thanks again for the suggestion! 🙏

Reply:
Added a `@typedef`! 🎉